internal/agent/device/hook/conditions.go

```diff
@@ -0,0 +1,192 @@
+// As a side-effect, it populates the command line variables of the action context with the corresponding list of files.
+func checkPathOpConditionForDir(cond api.HookConditionPathOp, actionCtx *actionContext) bool {
+	// ensure dir paths end with a trailing slash, so we don't accidentally match a file with a similar prefix
+	if len(cond.Path) < 1 || cond.Path[len(cond.Path)-1] != '/' {
+		cond.Path += "/"
+	}
+
+	conditionMet := false
+	if slices.Contains(cond.Op, api.FileOperationCreated) {
+		if treeFromPathContains(cond.Path, actionCtx.createdFiles) {
+			files := getContainedFiles(cond.Path, actionCtx.createdFiles)
+			appendFiles(actionCtx, FilesKey, files...)
+			appendFiles(actionCtx, CreatedKey, files...)
+			conditionMet = true
+		}
+	}
+	if slices.Contains(cond.Op, api.FileOperationUpdated) {
+		if treeFromPathContains(cond.Path, actionCtx.updatedFiles) {
+			files := getContainedFiles(cond.Path, actionCtx.updatedFiles)
+			appendFiles(actionCtx, FilesKey, files...)
+			appendFiles(actionCtx, UpdatedKey, files...)
+			conditionMet = true
+		}
+	}
+	if slices.Contains(cond.Op, api.FileOperationRemoved) {
+		if treeFromPathContains(cond.Path, actionCtx.removedFiles) {
+			files := getContainedFiles(cond.Path, actionCtx.removedFiles)
+			appendFiles(actionCtx, FilesKey, files...)
+			appendFiles(actionCtx, RemovedKey, files...)
+			conditionMet = true
+		}
+	}
+	if conditionMet {
+		actionCtx.commandLineVars[PathKey] = cond.Path
+	}
+	return conditionMet
+}
+
+// checkFileOpConditionForFile checks whether a specified operation (create, update, remove) has been performed
+// on the specified file.
+// As a side-effect, it populates the command line variables of the action context with the corresponding list of files.
+func checkPathOpConditionForFile(cond api.HookConditionPathOp, actionCtx *actionContext) bool {
+	conditionMet := false
+	if slices.Contains(cond.Op, api.FileOperationCreated) {
+		if pathEquals(cond.Path, actionCtx.createdFiles) {
+			appendFiles(actionCtx, FilesKey, cond.Path)
+			appendFiles(actionCtx, CreatedKey, cond.Path)
+			conditionMet = true
+		}
+	}
+	if slices.Contains(cond.Op, api.FileOperationUpdated) {
+		if pathEquals(cond.Path, actionCtx.updatedFiles) {
+			appendFiles(actionCtx, FilesKey, cond.Path)
+			appendFiles(actionCtx, UpdatedKey, cond.Path)
+			conditionMet = true
+		}
+	}
+	if slices.Contains(cond.Op, api.FileOperationRemoved) {
+		if pathEquals(cond.Path, actionCtx.removedFiles) {
+			appendFiles(actionCtx, FilesKey, cond.Path)
+			appendFiles(actionCtx, RemovedKey, cond.Path)
+			conditionMet = true
+		}
+	}
+	if conditionMet {
+		actionCtx.commandLineVars[PathKey] = cond.Path
+	}
+	return conditionMet
+}
+
+func pathEquals(path string, files map[string]ignv3types.File) bool {
+	_, ok := files[path]
+	return ok
+}
+
+func treeFromPathContains(path string, files map[string]ignv3types.File) bool {
+	for file := range files {
+		if strings.HasPrefix(file, path) {
```

Comment:
Just something to think about as it might not be an issue here but HasPrefix on a path would match `/etc/foo` and `/etc/foobar`.

Reply:
The caller ensures paths terminate with `/`, but I agree in terms of defensive programming I should hedge for this here, too. Will add.